FitGoal/FitGoal/Singup&Singin/CustomAuthenticator.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  CustomAuthenticator.swift
+//  FitGoal
+//
+//  Created by Eliany Barrera on 10/4/20.
+//  Copyright © 2020 Eliany Barrera. All rights reserved.
+//
+
+import Foundation
+
+class CustomAuthenticator {
+    var userName: UserInfoField?
+    var userEmail: UserInfoField?
+    var password: UserInfoField?
+    var passwordConfirmation: UserInfoField?
+    
+    var isUserInformationCorrect: Bool {
+        return isUserInfoStateValid(userInfoField: userName) &&
+            isUserInfoStateValid(userInfoField: userEmail) &&
+            isUserInfoStateValid(userInfoField: password) &&
+            isUserInfoStateValid(userInfoField: passwordConfirmation)
+    }
+    
+    private func isUserInfoStateValid(userInfoField: UserInfoField?) -> Bool {
+        if let field = userInfoField {
+            switch field.state {
+            case .valid:
+                return true
+            case .invalid:
+                return false
+            }
+        } else {
+            return false
+        }
+    }
+}
+
+enum UserInfoField {
```

Comment:
I'm not sure if adding this type adds anything here. If instead of this you have just a string field in the `CustomAuthenticator` class, you could have validation functions there, one for each field. What's the value of having this extra type?